### DISCUSSION TITLE
Scrolling to element doesn't work

### DISCUSSION THREAD

**Discussion by AdagioB** [quasarframework/quasar#13468] [General - Components / Directives / etc]
In my UMD project I have the following method:

```js
ScrollToTarget(elementName) {
  let el = this.$refs[elementName].$el;
  let target = Quasar.scroll.getScrollTarget(el);
  let offset = el.offsetTop;
  let duration = 300;
  Quasar.scroll.setVerticalScrollPosition(target, offset, duration);
},
```

And my component looks a bit like that (I hope I got everything relevant, I needed to cut quite a large section of irrelevant code)

```html
<template>
  <q-card
    class="q-mt-md q-mx-lg tableGroupColor"
    v-for="group in AllDataGroups"
    :key="group.id"
  >
    <q-card-section>`
      <q-table flat
        hide-bottom
	row-key="identID"
	class="tableGroupColor"
	separator="none"
	:rows="group.data"
	:columns="columns"
	:pagination.sync="pagination"
      >
        <template v-slot:header="props">
          <q-tr :props="props">
            <q-th v-for="col in props.cols"
              :key="col.name"`
              :props="props"`
              :class="GetTableHeaderClass(col.name)"
            >
              {{ col.label }}`
            </q-th>
          </q-tr>
        </template>

        <template v-slot:body="props">`
          <q-tr :ref="'pipeElement-' + props.row.identID + '-' + props.row.parentID">
            Some content
          </q-tr>
        </template>
      </q-table>
    </q-card-section>
  </q-card>
</template>
```

During a click on a button (not shown in the above code) the ScrollToTarget method is called, with elementName set to something like 'pipeElement-1-2' that fits with the ref of one of the <tr> segments above. The method finds the correct table (there could be more tables) and an offset, but it doesn't scroll all the way down to the element
I'm guessing the problem is with it being in tables. I tried the method on a page, where I had a lot of rows with buttons. When I called the method with the ref for the last button, it correctly scrolled there

Note that the table row I'm trying to scroll to is newly created. Maybe my code is executed before the element is shown? Is there a way to make it wait until everything has happened? 

Edit:
I have created a codepen that shows the situation. The bottoms that scrolls to other bottoms works perfectly fine, but the ones that scrolls to td or tr doesn't work

https://codepen.io/Adagio_B/pen/ZErxYJo?editors=1010

This is using newest version of Vue and quasar

**Comment by suleimanas** [comment]
try to ref the first td not tr

**Reply by AdagioB** [reply]
Tried it, but doesn't do any difference

**Comment by metalsadman** [comment]
template refs only available on mounted, a running codepen would help.

**Reply by AdagioB** [reply]
I have created a codepen that shows the situation. The bottoms that scrolls to other bottoms works perfectly fine, but the ones that scrolls to td or tr doesn't work

https://codepen.io/Adagio_B/pen/ZErxYJo?editors=1010

**Comment by pdanpdan** [comment]
The scrolling element in the table is inside of QTable, and in your case has no overflow, so there is nothing to scroll
